### PR TITLE
feat: go test and coverage

### PR DESCRIPTION
**Pull request by amornc**
About this orb

- This orb will run go test with covermode atomic then check quality gate from output.
- Input parameter testcoverage_threshold to specific the coverage percentage.
- save result to artifacts

### CODE REVIEW

File: src/commands/go-test.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# Descriptions should be short, simple, and clear.
+parameters:
+  testcoverage_threshold:
+    type: integer
+    default: 20
+    description: "the coverage percentage"
+  covermode:
+    type: string
+    default: "atomic"
+    description: "the coverage percentage"
```
Comment (puizeabix) [inline src/commands/go-test.yml:13]: incorrect description

File: pkg/db/db.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package db
```
Comment (puizeabix) [inline pkg/db/db.go:1]: Please remove all of snocko code from this repo krub

File: coverage.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import re;
```
Comment (puizeabix) [inline coverage.py:1]: Are we using this? or using the logic in the scripts/ folder. Seems to do the same thing, please remove if it's not used

**Review by puizeabix — CHANGES_REQUESTED** [review]
I've put some comments, please have a look

**Comment by amornc** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/zeabix-cloud-native/go-coverage-test-quality-gate?version=dev:7efeddb88b78b4d680e229eafa2d5cda35984467

**Comment by amornc** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/zeabix-cloud-native/go-coverage-test-quality-gate?version=dev:1bf2195d1b5040f9497d0d0ce2bb5ceff396ef0c

**Comment by amornc** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/zeabix-cloud-native/go-coverage-test-quality-gate?version=dev:14976d2a698e7bb58869e5f28fe1e873f433ece0

**Comment by amornc** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/zeabix-cloud-native/go-coverage-test-quality-gate?version=dev:4ed2968565e78719655e115f851df6b33a8aa2eb

File: src/commands/go-test.yml (hunk 1 above)
Comment (amornc) [reply src/commands/go-test.yml]: Done krub.

File: pkg/db/db.go (hunk 2 above)
Comment (amornc) [reply pkg/db/db.go]: Done krub.

File: coverage.py (hunk 3 above)
Comment (amornc) [reply coverage.py]: Already change the code to the new sample application krub.

**Comment by amornc** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/zeabix-cloud-native/go-coverage-test-quality-gate?version=dev:a457432158acc4e2da28285ffea21c7163f17275

**Comment by amornc** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/zeabix-cloud-native/go-coverage-test-quality-gate?version=dev:a457432158acc4e2da28285ffea21c7163f17275

**Pull request merged**